mesonpy/_editable.py

```diff
@@ -304,12 +304,15 @@ def rebuild(self) -> Node:
         env[MARKER] = os.pathsep.join((env.get(MARKER, ''), self._build_path))
 
         if self._verbose or bool(env.get(VERBOSE, '')):
-            print('+ ' + ' '.join(self._build_cmd))
-            stdout = None
+            # We do not want any output if there is no work to do. Code is adapted from:
+            # https://github.com/mesonbuild/meson/blob/a35d4d368a21f4b70afa3195da4d6292a649cb4c/mesonbuild/mtest.py#L1635-L1636
+            dry_run_build_cmd = self._build_cmd + ['-n']
+            p = subprocess.run(dry_run_build_cmd, cwd=self._build_path, env=env, capture_output=True)
+            if b'ninja: no work to do.' not in p.stdout and b'samu: nothing to do' not in p.stdout:
+                print('+ ' + ' '.join(self._build_cmd))
```

Comment:
I find the build command not very informative because it does not reflect which editable install is being rebuilt, nor does it state that `meson-python`'s editable install is in charge of making this build happen.

So maybe consider:

```suggestion
                module_names = " ".join(sorted(self._top_level_modules))
                print(f"meson-python: building {module_names}")
```

Here is the resulting user experience:

```
$ python -c "import sklearn"  # nothing to rebuild => no output
$ touch sklearn/ensemble/_gradient_boosting.pyx
$ python -c "import sklearn"
meson-python: building sklearn
[3/3] Linking target sklearn/ensemble/_gradient_boosting.cpython-311-darwin.so
```

Reply:
I like it!

Reply:
Two questions:
- do we want to keep the `ninja` build command for troubleshooting or are we fine not showing it?
- do we want to keep the `+` at the beginning of the first line? I guess maybe this was a marker to make it stand out in the output?

Reply:
I don't have an opinion on those points as long as the first message mentions "meson-python" and the name of the package(s) being rebuilt prior to displaying build outputs.